File: ios/App/Snappet/Features/Pomodoro/PomodoroSession.swift
import Foundation
import SwiftData

/// One completed focus block. Inserted into the shared store whenever a FOCUS phase
/// runs to completion; the integrator appends this type to `SnappetSchema.models`.
@Model
final class PomodoroSession {
    /// Length of the completed focus block, in minutes.
    var minutes: Int
    /// When the focus block finished.
    var completedAt: Date

    init(minutes: Int, completedAt: Date) {
        self.minutes = minutes
        self.completedAt = completedAt
    }
}

File: android/app/src/main/java/com/snappet/mobile/feature/pomodoro/PomodoroSession.kt
package com.snappet.mobile.feature.pomodoro

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * One completed focus block. Inserted into the shared store whenever a FOCUS phase runs to
 * completion. Mirrors the iOS `PomodoroSession` `@Model`.
 */
@Entity(tableName = "pomodoro_sessions")
data class PomodoroSession(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    /** Length of the completed focus block, in minutes. */
    val minutes: Int,
    /** When the focus block finished (epoch millis). */
    val completedAt: Long,
)
